### PR TITLE
Sidebar UX

### PR DESCRIPTION
**Pull request by shazm**
#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by enahum** [comment]
@shazm tests are failing and you have conflicts to resolve

/home/circleci/mattermost-mobile/app/components/channel_list/categories/categories.tsx
  4:9   error  'sortBy' is defined but never used  @typescript-eslint/no-unused-vars
  6:31  error  'Text' is defined but never used    @typescript-eslint/no-unused-vars

/home/circleci/mattermost-mobile/app/utils/mattermost_managed.ts
  27:5  error  Unexpected console statement  no-console

✖ 3 problems (3 errors, 0 warnings)

**Comment by shazm** [comment]
> @shazm tests are failing and you have conflicts to resolve
> 
> /home/circleci/mattermost-mobile/app/components/channel_list/categories/categories.tsx 4:9 error 'sortBy' is defined but never used @typescript-eslint/no-unused-vars 6:31 error 'Text' is defined but never used @typescript-eslint/no-unused-vars
> 
> /home/circleci/mattermost-mobile/app/utils/mattermost_managed.ts 27:5 error Unexpected console statement no-console
> 
> ✖ 3 problems (3 errors, 0 warnings)

I merged in the latest Gekidou, resolved some errors with the Podfile and just working through some of the TSC/ESLint errors.

**Comment by enahum** [comment]
@shazm can you please help update this PR as you mentioned https://github.com/mattermost/mattermost-mobile/pull/5928#issuecomment-1031146653

**Comment by shazm** [comment]
> @shazm can you please help update this PR as you mentioned https://github.com/mattermost/mattermost-mobile/pull/5928#issuecomment-1031146653

Yup almost there; will push fixes in the next hour or so!

File: app/actions/websocket/category.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if (!database) {
+        return;
+    }
+
+    // Add / Update Category
+    addOrUpdateCategories(serverUrl, [JSON.parse(msg.data.category)]);
+}
+
+export async function handleCategoryUpdatedEvent(serverUrl: string, msg: any) {
+    console.log('handleCategoryUpdatedEvent', msg);
```
Comment (avinashlng1080) [inline app/actions/websocket/category.ts:32]: Do we need the logs or can we remove all of them in this file?

**Review by avinashlng1080 — COMMENTED** [review]
LGTM.  Just a few comments here and there for my own understanding.

File: ios/Mattermost.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -739,43 +739,11 @@
 			);
 			inputPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-Mattermost/Pods-Mattermost-resources.sh",
-				"${PODS_ROOT}/../../node_modules/react-native-vector-icons/Fonts/AntDesign.ttf",
```
[the remark this answers is not in the document]
Comment (shazm) [reply ios/Mattermost.xcodeproj/project.pbxproj:742]: IIRC we only use the MM CompasIcons.

File: app/actions/websocket/index.ts [hunk 3]
```diff
@@ -302,6 +303,19 @@ export async function handleEvent(serverUrl: string, msg: WebSocketMessage) {
         case WebsocketEvents.APPS_FRAMEWORK_REFRESH_BINDINGS:
             break;
 
+        case WebsocketEvents.CATEGORY_CREATED:
```
[the remark this answers is not in the document]
Comment (shazm) [reply app/actions/websocket/index.ts:306]: I certainly see your point, but WS (and remote data fetching) is what ends up resulting in UX changes. I can split this up more in that case? Maybe have:


- WS Handling PR
- Remote Category Fetching PR
- Sidebar UX PR

If that works? cc @enahum

File: app/components/channel_list/search/index.tsx [hunk 4]
```diff
@@ -2,7 +2,8 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {StyleSheet, TextInput, View} from 'react-native';
+import {StyleSheet, View} from 'react-native';
+import {TextInput} from 'react-native-gesture-handler';
```
[the remark this answers is not in the document]
Comment (shazm) [reply app/components/channel_list/search/index.tsx:6]: Ah whoops, will fix - should just be RN.

File: app/constants/index.ts [hunk 5]
```diff
@@ -42,7 +41,6 @@ export {
     Post,
     PostDraft,
     Preferences,
-    Profile,
```
[the remark this answers is not in the document]
Comment (shazm) [reply app/constants/index.ts:45]: Hmm not sure - might just be a relic from the gekidou merge. Will revert.

File: app/components/channel_list/index.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@constants/database';
+
+const {SERVER: {SYSTEM}} = MM_TABLES;
```
[the remark this answers is not in the document]
Comment (shazm) [reply app/components/channel_list/index.ts:11]: I use the ESLint / Prettier auto-formatter; which saves it to this - what order should it be?

File: app/components/channel_list/categories/body/channel/channel_list_item.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        <TouchableOpacity onPress={() => switchToChannelById(serverUrl, myChannel.id)}>
+            <View style={styles.container}>
+                <ChannelIcon
+                    shared={false}
+                    type={channel.type}
+                    size={24}
+                />
+                <Text
+                    style={[
+                        bright ? textStyle.bright : textStyle.regular,
```
[the remark this answers is not in the document]
Comment (shazm) [reply app/components/channel_list/categories/body/channel/channel_list_item.tsx:68]: `textStyle` is already a `StyleSheet` no? I thought that avoided the need for `useMemo`.

**Pull request closed without merging**